http_util.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connect
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+// WriteError should be called from HTTP middlewares to handle connect errors. It
+// writes a status code encodes a connect error as JSON,  and writes to the response.
+func WriteError(responseWriter http.ResponseWriter, err error) error {
```

Comment:
@akshayjshah originally had a request in here, but I didn't see any reason to include it. Potentially it was there to close the request body, and when I lifted this code from https://github.com/bufbuild/connect-go/blob/main/protocol_connect.go#L553 I did see that function closing the request. But when looking into https://pkg.go.dev/net/http#Request.Close I believe that `WriteError` doesn't need to close the request if the server will do so anyway.